rust/connlib/tunnel/src/client.rs

```diff
@@ -797,7 +836,10 @@ impl ClientState {
             .longest_match(destination)
             .map(|(_, res)| res.id);
 
-        maybe_dns_resource_id.or(maybe_cidr_resource_id)
+        // We need to filter disabled resources because we never remove resources from the stub_resolver
```

Comment:
From what I can see, we now do?

Reply:
We never do from the `ips_to_fqdn` map, I think we don't so we always get the same ip if a resource gets re-added at a later time.

But I think a better long-time solution would be to separate that map into `ip->fqdn` and `fqdn->resource_id` and we can remove the resource from the latter one?

Reply:
> But I think a better long-time solution would be to separate that map into `ip->fqdn` and `fqdn->resource_id` and we can remove the resource from the latter one?

That was a deliberate performance optimisation so we don't have to pattern match for the right DNS resource on every packet.

Reply:
If we keep a map `fqdn->resource_id` we don't need to pattern match, it's a single look up

Reply:
> If we keep a map `fqdn->resource_id` we don't need to pattern match, it's a single look up

Yes that is true.

Reply:
I'll create a follow up PR with this change where we can discuss it more